Tasks/AzureAppServiceManage/azureappservicemanage.ts

```diff
@@ -5,13 +5,16 @@ var azureRmUtil = require('azurerest-common/azurerestutility.js');
 var kuduLogUtil = require('azurerest-common/kududeploymentstatusutility.js');
 
 async function swapSlot(endPoint, resourceGroupName: string, webAppName: string, sourceSlot: string, swapWithProduction: boolean, targetSlot: string, preserveVnet: boolean) {
-    if(swapWithProduction) {
-        targetSlot = "production";
+    try {
+        var isSlotSwapSuccess = await azureRmUtil.swapWebAppSlot(endPoint, resourceGroupName, webAppName, sourceSlot, targetSlot, preserveVnet);
+        if (isSlotSwapSuccess) {
+            tl._writeLine(tl.loc("Successfullyswappedslots", webAppName, sourceSlot, targetSlot));
+        }
     }
-    if(sourceSlot === targetSlot){
-        throw new Error(tl.loc("SourceAndTargetSlotCannotBeSame"));
+    catch (errorMessage) {
+        tl.error(errorMessage);
```

Comment:
Do you need to do tl.error if you are throwing the error again. In that case you wont need the try-catch block in this function